chia/wallet/wallet_state_manager.py

```diff
@@ -1602,7 +1603,8 @@ async def add_coin_states(
         fork_height: Optional[uint32],
     ) -> bool:
         try:
-            await self._add_coin_states(coin_states, peer, fork_height)
+            unique_coin_states = sort_coin_states(list(set(coin_states)))
+            await self._add_coin_states(unique_coin_states, peer, fork_height)
```

Comment:
This isn't the right place to deduplicate the coin states imo because this is after we already sorted all coin states here https://github.com/Chia-Network/chia-blockchain/blob/b8a1515d7350f92e246a369b2685fafdc7864d5c/chia/wallet/wallet_node.py#L870 before we split them up into batches to process in tasks which call `add_coin_states` here https://github.com/Chia-Network/chia-blockchain/blob/b8a1515d7350f92e246a369b2685fafdc7864d5c/chia/wallet/wallet_node.py#L905 

So basically you now only deduplicate the separate batches instead the whole list coming from a peer and we would still validate duplicates and also this adds a second round of sorting.

Reply:
I just created https://github.com/Chia-Network/chia-blockchain/pull/15598.